src/ReinforcementLearningCore/src/policies/agent/multi_agent.jl

```diff
@@ -188,13 +188,24 @@ function Base.push!(multiagent::MultiAgentPolicy, ::PreActStage, env::E) where {
     end
 end
 
+# Multiagent Version
+function RLBase.plan!(agent::Agent{P,T,C}, env::E, p::Symbol) where {P,T,C,E<:AbstractEnv}
+    action = RLBase.plan!(agent.policy, env, p)
+    push!(agent.cache, action)
+    action
+end
+
 # Like in the single-agent case, push! at the PostActStage() calls push! on each player with the reward and termination status of the environment
 function Base.push!(multiagent::MultiAgentPolicy, ::PostActStage, env::E) where {E<:AbstractEnv}
     for player in players(env)
-        push!(multiagent[player].cache, reward(env, player), is_terminated(env))
+        push!(agent.trajectory, multiagent[player].cache, reward(env, player), is_terminated(env))
```

Comment:
When the environment is simultaneous, this is totally fine. 
If you think of sequential environments, this is never called, but the reward has to be collected by reward(env, player). Is this a problem or do we assume, that the multi_agent environment handles this?
Also, agent.trajectory has to be replaced by multiagent[player].trajectory in order to work.